include/accessor.h

```diff
@@ -221,17 +222,21 @@ class accessor<DataT, Dims, Mode, target::device> : public detail::accessor_base
 	friend struct detail::accessor_testspy;
 
   public:
+	CELERITY_DETAIL_HACK_CLANG_TIDY_ALLOW_NON_CONST(accessor)
```

Comment:
I know this macro was my idea, but maybe we can make the connection between the magic happening here and the explicitly-defaulted constructors more obvious.

- Either by renaming it, e.g. `CELERITY_DETAIL_HACK_DEFINE_COPY_CTOR_TO_ALLOW_CLANG_TIDY_NON_CONST` (that's a mouthful)
- Or by just substituting the macro invocations with the actual constructor plus a comment

What do you think?